test/nats.test.ts

```diff
@@ -3,11 +3,21 @@ import {it} from 'mocha'
 import {expect} from 'chai'
 
 describe('Nat tests', () => {
-  it('should return a NoArbitrary if the bounds are invalid', () => {
+  it('should return a valid range if min < 0', () => {
     expect(
       fc.scenario()
         .forall('a', fc.integer(Number.MIN_SAFE_INTEGER, -1))
         .forall('b', fc.integer(0, Number.MAX_SAFE_INTEGER))
+        .then(({a, b}) => fc.nat(a, b).sample(10).every(i => i.value >= 0))
+        .check()
+    ).to.have.property('satisfiable', true)
+  })
+
+  it('should return a NoArbitrary if max < 0', () => {
```

Comment:
What if max < min?

Reply:
I think maybe we could do something that would implicitly test what you said. Currently, we have `max < 0 ? NoArbitrary : new ArbitraryInteger(Math.max(0, min), max)`. My suggestion is to change that line of code to `max < 0 ? NoArbitrary : integer(Math.max(0, min), max)`.

Reply:
The composition of `nat` over `integer` looks good to me, @RuiGuedes. Go ahead :)